lua/gen/init.lua

```diff
@@ -54,12 +54,28 @@ end
 M.command = 'ollama run $model $prompt'
 M.model = 'mistral:instruct'
 
+local function is_serving(pname)
+    local handle, ok = pcall(io.popen, 'ps aux | grep "' .. pname .. '" | grep -v grep')
```

Comment:
Why not use pgrep
```suggestion
    local handle, ok = pcall(io.popen, 'pgrep "' .. pname .. '"')
    if not ok or not handle then
      return false
    end
    handle:close()
    return true
```

This should work on both gnu and bsd systems, and you don't need to read the output